roerich/algorithms/calc_metrics.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABCMeta, abstractmethod
+import numpy as np
+import pandas as pd
+from sklearn.metrics import pairwise_distances, roc_curve, roc_auc_score
+from sklearn.model_selection import StratifiedKFold, KFold, train_test_split
+from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+from copy import deepcopy
+from joblib import Parallel, delayed
+from scipy import interpolate
+from scipy.signal import argrelmax
+
+from .cpdc import ChangePointDetectionBase
+from roerich.scores.fd import frechet_distance
+from roerich.scores.mmd import maximum_mean_discrepancy
+
+
+class ScoreCalculator(ChangePointDetectionBase):
+
+    def __init__(self, metric=None, func=None, periods=1, window_size=100, step=1, n_runs=1):
+        super().__init__(periods=periods, window_size=window_size, step=step, n_runs=n_runs)
+        self.metric = metric
+        self.func = func
+
+    def reference_test_predict(self, X_ref, X_test):
+
+        if self.metric == "EnergyDist":
+            n = X_ref.shape[0]
+            E = 2*pairwise_distances(X_ref, X_test, metric='euclidean') - pairwise_distances(X_test, metric='euclidean') - pairwise_distances(X_ref, metric='euclidean')
```

Comment:
@maxslamdunk, hi!
Let's implements this function like it was done for fd and mmn in https://github.com/HSE-LAMBDA/roerich/tree/main/roerich/scores